### PR TITLE
Add vast-client via single package.json w/ npm auto-update

### PR DESCRIPTION
**Pull request by kirintwn**
Pull request for issue: #12210 

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/dailymotion/vast-client-js
 * Official website (optional, not the repository):
 * NPM package url (optional): https://www.npmjs.com/package/vast-client
 * License and its reference: MIT
 * GitHub / Bitbucket popularity (required):
   - Count of watchers: 51
   - Count of stars: 186
   - Count of forks: 127
 * NPM download stats (optional):
   - Downloads in the last day: 38
   - Downloads in the last week: 168
   - Downloads in the last month: 842

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [x] No already exist / duplicated issue and PR
 * [x] The lib has notable popularity
   * [ ] More than 200 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [x] More than 800 downloads stats per month on npm registry
 * [x] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [x] Has valid tags for each versions (for git auto-update)
 * [x] Auto-update setup
 * [x] Auto-update target/source is valid.
 * [x] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

File: ajax/libs/vast-client/package.json [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  },
+  "license": "MIT",
+  "autoupdate": {
+    "source": "git",
+    "target": "git://github.com/dailymotion/vast-client-js.git",
+    "fileMap": [
+      {
+        "basePath": "",
+        "files": [
+          "*.js"
```
Comment (sashberd) [inline ajax/libs/vast-client/package.json]: Lets use vast-client*.(js|map) pattern

File: ajax/libs/vast-client/package.json [hunk 2]
```diff
@@ -0,0 +1,32 @@
+{
+  "name": "vast-client",
+  "filename": "vast-client.js",
```
Comment (sashberd) [inline ajax/libs/vast-client/package.json]: After you will minify the file use the min.js here

**Comment by cy91244** [comment]
@sashberd I have done the changes, please confirm, thanks.

File: ajax/libs/vast-client/package.json [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  },
+  "license": "MIT",
+  "autoupdate": {
+    "source": "git",
+    "target": "git://github.com/dailymotion/vast-client-js.git",
+    "fileMap": [
+      {
+        "basePath": "",
+        "files": [
+          "*.(js|map)"
```
Comment (sufuf3) [inline ajax/libs/vast-client/package.json]: This pattern can't get any files.
Let's modify to `"vast-client*"`.
PS. You could try auto-update tool to check.

Many thanks.

File: ajax/libs/vast-client/package.json [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    "ad",
+    "advertising",
+    "iab",
+    "in-stream",
+    "video"
+  ],
+  "author": "Olivier Poitrey <rs@dailymotion.com>",
+  "repository": {
+    "type": "git",
+    "url": "git://github.com/dailymotion/vast-client-js"
```
Comment (sufuf3) [inline ajax/libs/vast-client/package.json]: Let's modify as the same with upstream.
`"url": "https://github.com/dailymotion/vast-client-js"`
https://github.com/dailymotion/vast-client-js/blob/v1.8.0/package.json#L16
Thanks

**Comment by sashberd** [comment]
@sufuf3 I helped with this one. FYR

**Comment by sashberd** [comment]
@extend1994, @PeterDaveHello  The current autoupdate fetch v2.0.0, but not v2.0.1 and v2.0.2. Should I add 

```
{
        "basePath": "dist",
        "files": [
          "*/vast-client*"
        ]
      }
```
to autoupdate config?

**Comment by sashberd** [comment]
@xtuc Done. FYR

File: ajax/libs/vast-client/package.json [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    "video"
+  ],
+  "author": "Olivier Poitrey <rs@dailymotion.com>",
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/dailymotion/vast-client-js"
+  },
+  "license": "MIT",
+  "autoupdate": {
+    "source": "git",
```
Comment (MattIPv4) [inline ajax/libs/vast-client/package.json]: I would prefer to use an npm auto-update config instead of git here, please :)

**Comment by sashberd** [comment]
@MattIPv4 Moved to npm

**Pull request merged**